source/driver-vs-mongoose.txt

```diff
@@ -0,0 +1,142 @@
+them as it is similar to using an Object-Relational Mapper, or ORM,
+which they've likely used when working with a relational database.
+
+The official MongoDB native driver provides direct low-level access to
+MongoDB, allows for more flexibility and control but requires more
+manual effort for data modeling and validation. On the other hand,
+Mongoose abstracts away many complexities, providing data modeling,
+schema validation, middleware support, and other convenient features,
+making it easier to work with MongoDB in Node.js applications. The
+choice between the two depends on your specific project requirements and
+how much control and flexibility you need in database interactions.
+
+What is Mongoose?
+-----------------
+
+Mongoose is an Object-Document Mapping (ODM) library for Node.js,
+specifically designed to work with MongoDB. It provides a higher-level
+abstraction over MongoDB's native driver, simplifying interactions with
+the database. With Mongoose, you can define data schemas and models,
+allowing you to work with MongoDB in a more structured and intuitive
+manner. 
+
+Differences in Schema Management
+--------------------------------
+
+Both the {+driver-long+} and Mongoose let you apply schema validation
+to your collection. However, the biggest difference between the two are
+the way they each set and manage schema validation.
+
+Mongoose allows you to define data schemas using its schema notation.
+These schemas define the structure of the documents to be stored in
+MongoDB, along with data types, default values, and validation rules.
+Mongoose provides built-in support for data validation based on the
+defined schema rules. It ensures that the data being saved to the
+database adheres to the specified schema, preventing invalid data from
+being stored. 
+
+The {+driver-long+} applies the schema validation at the database
+level. In comparison, Mongoose applies schema validation at the application level. In the
+{+driver-short+} schema validation works on individual collections by
+using either `JSON Schema Validation <https://www.mongodb.com/docs/manual/core/schema-validation/specify-json-schema/>`__
+or `Schema Validation with Query Operators <https://www.mongodb.com/docs/manual/core/schema-validation/specify-query-expression-rules/>`__.
+Validation with query operators is useful for dynamic validation rules
+calculated at runtime. By contrast, JSON validation is useful for, among other
+reasons, being an open standard which makes it easy to connect to
+third-party libraries.
+
+Similarities
+------------
+
+As Mongoose is built upon the {+driver-long+}, they share many
+capabilities. Even if you use Mongoose, you always have the option to
+use any native function or command built into the {+driver-short+}
+alongside Mongoose. Mongoose also offers its own methods that are quite
+similar to the {+driver-short+}, such as ``updateOne()`` and
+``replaceOne()``.
```

Comment:
```suggestion
If you use Mongoose, you can also integrate any APIs from the {+driver-short+}.
Mongoose also offers its own APIs that are similar to those provided by the driver, including ways to update and replace documents.
```

Reply:
@rustagir in this case since API similarities are being called out wouldn't it be better to keep the methods in there (`updateOne()`) as opposed to generalizing it to "ways to update ..."?